processing/manual_correction.sh

```diff
@@ -0,0 +1,38 @@
+#!/bin/bash
```

Comment:
we should also create a script for the vertebral labeling, or even better: include manual labeling inside this script. You can figure out if the input file is designed as a segmentation or a labeling, by looking at the end of the file (seg vs. label)

Reply:
Okay, I will implement it.

Reply:
One question before I start with implementation - `$FILES` variable now contains filenames of background images, not filenames of segmentation (or labeling), so I cannot figure out if to open segmentation or labeling. So, do you agree with passing filenames of segmentation/labeling into `$FILES` variable instead of current background image filenames?

Reply:
> One question before I start with implementation - `$FILES` variable now contains filenames of background images, not filenames of segmentation (or labeling), so I cannot figure out if to open segmentation or labeling. So, do you agree with passing filenames of segmentation/labeling into `$FILES` variable instead of current background image filenames?

great point! so, i see 2 choices:
- Choice 1: create two variables: $FILES_SEG and $FILES_LABEL
  - Pros: easy copy/paste from the QC window (ie: no need to manually type "_seg" or "_label", which is prone to error)
  - Pros: no need to parse the file to retrieve the file name of the image
  - Pros: if people do the QC of segs, and then of labeling, it might be easier to list the files into two separate variables 
  - Pros: makes code cleaner (i.e. we would have two loops: one for $FILES_SEG that calls FSLeyes, and one for $FILES_LABEL that calls sct_label_utils)
- Choice 2: create a single $FILES, that list the segmentations or labels to modify
  - Pros: might be easier for the user
  - Pros: more intuitive (user points the problematic segmentation/labels), however does require manual modification of file names (humans make mistakes)
  - Cons: code less clean (you will need to parse the file, figure out if it is a seg or a label, and then send to the appropriate if case)

Reply:
Thanks for brainstorming! So, I would vote for the first choice. It eliminates manual adding of "_seg" or "_label" suffix which I can see as the biggest disadvantage and possible prone to errors. I'll implement choice 1.

Reply:
Hm, I have encountered on problem when I tried to pass two list (array in language of bash) variables (`$FILES_SEG` and `$FILES_LABEL`) as inputs to bash script. It looks that bash does not allow to pass arrays, see [here](https://stackoverflow.com/questions/16461656/how-to-pass-array-as-an-argument-to-a-function-in-bash) and [here](https://askubuntu.com/questions/674333/how-to-pass-an-array-as-function-argument).

Workaround could be to pass only one list/array variable as input (so, the choice 2 from previous brainstorming) and use something like this `INPUTS=$@` inside the script to catch all elements of input array. But still here would be disadvantage that script would have to be called `manual_correction.sh ${FILES_SEG[@]}` instead of simply `manual_correction.sh $FILES_SEG`.

Another workaround could be to define input variables as simple strings, pass them to script and split them into individual filenames inside the script.

Reply:
how about simply passing a json and/or yaml file? another pros is that the information is saved

Reply:
It is another option. Or simply planar text file with individual filenames as rows. This file would have to be added to .gitignore to do not conflict if user save it into repo folder, however (probably not problem). But sure, saving of this information is huge pros.

Reply:
> It is another option. Or simply planar text file with individual filenames as rows. 

no, this is not following proper convention. This is why json and yaml formats exist (universal convention).

> This file would have to be added to .gitignore to do not conflict if user save it into repo folder

personally i've been launching those script from within the output folder (not from the repos).

but yes, people can add it to their .gitignore if they want to

Reply:
p.s. yaml is way friendlier than json

Reply:
Okay, yaml looks fine. Which exact yaml format we will use?
inline with comments and without keys
```
---
# FILES_SEG
[sub-amu01_T1w_RPI_r.nii.gz, sub-amu01_T1w_RP_r.nii, sub-amu01_T2w_RPI_r.nii.gz]
# FILES_LABEL
[sub-amu01, sub-amu02]
```
or inline with keys

```
---
FILES_SEG: [sub-amu01_T1w_RPI_r.nii.gz, sub-amu01_T1w_RP_r.nii, sub-amu01_T2w_RPI_r.nii.gz]
FILES_LABEL: [sub-amu01, sub-amu02]
```

Reply:
We probably want the key as it will be used for choosing the right pipeline, no?
Also, I’d show one subject per line (PEP8)

Reply:
Yeah, option with keys is probably more cleaner. However, I was able to identify pipeline also based on comment above certain line.
So, yaml file with keys and one subject perline would look:

```
---
FILES_SEG:
- sub-amu01_T1w_RPI_r.nii.gz
- sub-amu01_T1w_RP_r.nii
- sub-amu01_T2w_RPI_r.nii.gz
FILES_LABEL:
- sub-amu01
- sub-amu02
```

Reply:
beautiful! now we just need a yaml parser 😊

Reply:
Added here https://github.com/sct-pipeline/spine-generic/pull/87/commits/58e4224f883a89f7e852002a1023689557fe36f3 😊